lib/services/authentication.go

```diff
@@ -31,13 +31,22 @@ import (
 	"github.com/gravitational/teleport/lib/utils"
 )
 
+// This is bcrypt hash for password "barbaz".
+var fakePasswordHash = []byte(`$2a$10$Yy.e6BmS2SrGbBDsyDLVkOANZmvjjMR890nUGSXFJHBXWzxe7T44m`)
+
 // ValidateLocalAuthSecrets validates local auth secret members.
 func ValidateLocalAuthSecrets(l *types.LocalAuthSecrets) error {
+	var hash []byte
 	if len(l.PasswordHash) > 0 {
-		if _, err := bcrypt.Cost(l.PasswordHash); err != nil {
-			return trace.BadParameter("invalid password hash")
-		}
+		hash = l.PasswordHash
+	} else {
+		// Prevent timing attacks by using a fake hash.
+		hash = fakePasswordHash
```

Comment:
Why is this necessary now? Was this always necessary and we just never bothered?

Reply:
What are we trying to accomplish here? Its' not clear to me what the intent is.

Reply:
@codingllama I think it was a miscommunication on my side, I was under impression that either you or perhaps Mike Jensen claimed that this branch should also be protected from timing attacks. I don't really think myself that it's necessary, so I'm removing this part.